data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/JsonTypeDefinitionBeaconRestApi.java

```diff
@@ -13,15 +13,236 @@
+import tech.pegasys.teku.beaconrestapi.handlers.v1.node.GetPeerById;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.node.GetPeerCount;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.node.GetPeers;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.node.GetSyncing;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.node.GetVersion;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.GetAggregateAttestation;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.GetAttestationData;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.GetNewBlindedBlock;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.GetProposerDuties;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.GetSyncCommitteeContribution;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostAggregateAndProofs;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostAttesterDuties;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostContributionAndProofs;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostPrepareBeaconProposer;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostRegisterValidator;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostSubscribeToBeaconCommitteeSubnet;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostSyncCommitteeSubscriptions;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostSyncDuties;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.validator.PostValidatorLiveness;
+import tech.pegasys.teku.beaconrestapi.handlers.v2.debug.GetChainHeadsV2;
+import tech.pegasys.teku.beaconrestapi.handlers.v2.debug.GetState;
+import tech.pegasys.teku.beaconrestapi.handlers.v2.validator.GetNewBlock;
+import tech.pegasys.teku.infrastructure.http.ContentTypeNotSupportedException;
+import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
+import tech.pegasys.teku.infrastructure.restapi.RestApi;
+import tech.pegasys.teku.infrastructure.restapi.RestApiBuilder;
+import tech.pegasys.teku.infrastructure.version.VersionProvider;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionCache;
+import tech.pegasys.teku.storage.client.ChainDataUnavailableException;
+import tech.pegasys.teku.validator.api.NodeSyncingException;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider;
+
 public class JsonTypeDefinitionBeaconRestApi implements BeaconRestApi {
+  private final RestApi restApi;
+
+  public JsonTypeDefinitionBeaconRestApi(
+      final BeaconRestApiConfig config,
+      final DataProvider dataProvider,
+      final Eth1DataProvider eth1DataProvider,
+      final Spec spec) {
+    restApi = create(config, dataProvider, eth1DataProvider, spec);
+  }
 
   @Override
   public void start() {
-    throw new UnsupportedOperationException("Unsupported start operation");
+    restApi.start().ifExceptionGetsHereRaiseABug();
```

Comment:
We should change `start` (and `stop`) so that it returns a `SafeFuture`.  The old implementation can just return `SafeFuture.COMPLETED` but where this is called in `BeaconChainController` it's actually creating a `SafeFuture` from the call:

```java
return initialize()
        .thenCompose(
            (__) -> SafeFuture.fromRunnable(() -> beaconRestAPI.ifPresent(BeaconRestApi::start)));
```

so that would just become:
```java
return initialize()
        .thenCompose(
            __ -> beaconRestAPI.map(BeaconRestApi::start).orElse(SafeFuture.COMPLETE)));
```

Reply:
Okay I have tried that, and am not sure what to do in the `orElse()` as returning a `SafeFuture<Void>` is not allowed.
<img width="442" alt="Screen Shot 2022-07-01 at 2 16 53 PM" src="https://user-images.githubusercontent.com/45641759/176810341-6f270a86-b5b2-4cb7-baf6-898384c7729d.png">